tasks/gridxc-multi.yml

```
@@ -0,0 +1,94 @@
+#
+# Installation of libgridxc (serial and MPI versions, double precision)
+#
+# In ubuntu 16, the libxc-dev package does not have a fortran library
+#
+- name: Install optional libxc dependency
+  become: true
+  apt:
+    name:
+    - libxc-dev
+  when: ansible_facts['distribution_major_version'] | int > 16
+  register: libxc_installed
+
+- name: Set libxc incantation
+  set_fact:
+    libgridxc_libxc_incantation: "--with-libxc='-I/usr/include,-L/usr/lib/x86_64-linux-gnu -lxcf90 -lxc'"
+  when: ansible_facts['distribution_major_version'] | int > 16
+
+- name: Leave libxc incantation empty
+  set_fact:
+    libgridxc_libxc_incantation: ""
+  when: ansible_facts['distribution_major_version'] | int <= 16
+
+
+- name: Get libgridxc source
+  get_url:
+    url: "{{ libgridxc_url }}"
+    dest: "{{ libgridxc_code_folder }}/{{ libgridxc_src_archive }}"
+  register: libgridxc_download
+
+- name: Extract libgridxc source
+  unarchive:
+    src: "{{ libgridxc_download.dest }}"
+    dest: "{{ libgridxc_code_folder }}"
+    remote_src: true
+
+- name: Create config directories
+  file:
+    path: "{{ libgridxc_topdir }}/{{ item }}"
+    state: directory
+  with_items:
+  - build_dp
+  - build_mpi_dp
+
+#
+# The following could also be parametrized with 'items' in a loop,
+# using appropriate variables for 'dp' and 'mpi_dp'.
+#
+# --------- dp version
+
+- name: Configure libgridxc dp
+  shell: " ../configure --enable-multiconfig  --without-mpi {{ libgridxc_libxc_incantation }} --prefix={{ libgridxc_prefix }}"
+  args:
+    chdir: "{{ libgridxc_topdir }}/build_dp"
+    creates: "{{ libgridxc_topdir }}/build_dp/Makefile"
+  register: libgridxc_dp_configure
+
+- name: Make libgridxc dp
+  shell: "make"
+  args:
+    chdir: "{{ libgridxc_topdir }}/build_dp"
+    creates: "{{ libgridxc_topdir }}/build_dp/src/libgridxc_dp.la"
+  register: libgridxc_dp_make
+
+- name: Install libgridxc dp
+  shell: make install
+  args:
+    chdir: "{{ libgridxc_topdir }}/build_dp"
+    creates: "{{ libgridxc_prefix }}/lib/libgridxc_dp.a"
+  become: yes
+  register: libgridxc_dp_install
+
+# --------- mpi_dp version
+- name: Configure libgridxc mpi_dp
+  shell: "MPIFC=mpif90 MPICC=mpicc ../configure --enable-multiconfig --with-mpi {{ libgridxc_libxc_incantation }} --prefix={{ libgridxc_prefix }}"
```

you could just use

```suggestion
  shell: "../configure --enable-multiconfig --with-mpi {{ libgridxc_libxc_incantation }} --prefix={{ libgridxc_prefix }}"
  environment:
  	MPIFC: mpif90
  	MPICC: mpicc
```

but not strictly necessary